priv/perf/apps/load_test/lib/service/faucet.ex

```diff
@@ -45,48 +69,122 @@ defmodule LoadTest.Service.Faucet do
+      )
 
-    state = struct!(__MODULE__, account: faucet, fee: fee_wei, utxos: %{@eth => eth_utxo})
     {:ok, state}
   end
 
-  def handle_call(:get_faucet, _from, %__MODULE__{account: faucet} = state) do
-    {:reply, {:ok, faucet}, state}
+  def handle_call(:get_faucet, _from, %__MODULE__{faucet_account: faucet_account} = state) do
+    {:reply, {:ok, faucet_account}, state}
   end
 
-  def handle_call({:fund_child_chain, account, amount, @eth = token}, _from, %__MODULE__{account: faucet} = state) do
-    {utxo, utxo_amount} = state.utxos[token]
-    change = utxo_amount - amount - state.fee
-    if change < 0, do: raise({:error, :change_below_zero})
+  def handle_call(
+        {:fund_child_chain, receiver, amount, currency},
+        _from,
+        %__MODULE__{
+          faucet_account: faucet_account
+        } = state
+      ) do
+    utxo = check_sufficient_funds(state, currency, amount)
 
-    Logger.debug("Funding user: #{Encoding.to_hex(account.addr)} with UTXO: #{Utxo.pos(utxo)}")
+    change = utxo.amount - amount - state.fee
+    if change < 0, do: raise({:error, :insufficient_faucet_funds})
 
     outputs = [
-      %Utxo{amount: change, currency: token, owner: faucet.addr},
-      %Utxo{amount: amount, currency: token, owner: account.addr}
+      %Utxo{amount: change, currency: currency, owner: faucet_account.addr},
+      %Utxo{amount: amount, currency: currency, owner: receiver.addr}
     ]
 
-    {:ok, blknum, txindex} = Transaction.submit_tx([utxo], outputs, [faucet], @fund_child_chain_account_retries)
-    {:ok, change_utxo} = Utxo.new(%{blknum: blknum, txindex: txindex, oindex: 0})
-    {:ok, user_utxo} = Utxo.new(%{blknum: blknum, txindex: txindex, oindex: 1, amount: amount})
+    Logger.debug("Funding user #{Encoding.to_hex(receiver.addr)} with #{amount} from utxo: #{Utxo.pos(utxo)}")
+    {:ok, blknum, txindex} = Transaction.submit_tx([utxo], outputs, [faucet_account], @fund_child_chain_account_retries)
 
-    updated_state = Map.put(state, :utxos, %{state.utxos | token => {change_utxo, change}})
+    next_faucet_utxo = %Utxo{blknum: blknum, txindex: txindex, oindex: 0, amount: change}
+    user_utxo = %Utxo{blknum: blknum, txindex: txindex, oindex: 1, amount: amount}
+
+    updated_state = Map.put(state, :utxos, Map.put(state.utxos, currency, next_faucet_utxo))
 
     {:reply, {:ok, user_utxo}, updated_state}
   end
 
-  defp get_funded_faucet_account(opts) do
-    faucet_private_key = Keyword.fetch!(opts, :faucet_private_key)
-    {:ok, faucet} = Account.new(faucet_private_key)
+  @spec check_sufficient_funds(state(), Utxo.address_binary(), pos_integer()) :: Utxo.t()
+  defp check_sufficient_funds(
+         %{utxos: utxos, faucet_account: faucet_account, fee: fee},
+         currency,
+         amount
+       ) do
+    utxo =
+      case Map.has_key?(utxos, currency) do
+        true ->
+          utxos[currency]
+
+        _ ->
+          get_utxos(faucet_account.addr)
```

Comment:
```suggestion
          faucet_account.addr
          |> get_utxos()
          |> get_largest_utxo_by_currency(faucet_account, currency)
```
when using pipes, you try to start them with a raw value.
but restrain yourself in using pipes if you're not chaining 1 or more functions.